src/main/java/com/myhome/services/springdatajpa/UserSDJpaService.java

```diff
@@ -39,33 +40,31 @@
  */
 @Service
 @Slf4j
+@RequiredArgsConstructor
 public class UserSDJpaService implements UserService {
 
   private final UserRepository userRepository;
   private final UserMapper userMapper;
   private final PasswordEncoder passwordEncoder;
-
-  @Autowired
-  private CommunityService communityService;
-
-  public UserSDJpaService(UserRepository userRepository,
-      UserMapper userMapper,
-      PasswordEncoder passwordEncoder) {
-    this.userRepository = userRepository;
-    this.userMapper = userMapper;
-    this.passwordEncoder = passwordEncoder;
-  }
+  private final CommunityService communityService;
 
   @Override public UserDto createUser(UserDto request) {
     generateUniqueUserId(request);
     encryptUserPassword(request);
     return createUserInRepository(request);
   }
 
-  @Override public Set<User> listAll() {
+  @Override public Set<User> listAll(Integer limit, Integer start) {
     Set<User> userListSet = new HashSet<>();
     userRepository.findAll().forEach(userListSet::add);
-    return userListSet;
+    Stream<User> userStream = userListSet.stream();
+    if (start != null) {
+      userStream = userStream.filter(user -> user.getUserId().startsWith(String.valueOf(start)));
```

Comment:
Implementation of `limit` seems to be fine. But, start ideally has to be used for pagination. Let me try to explain it with a use case.

User hits `/users?limit=20&start=0` - Display the first 20 users.
User hits `/users?limit=20&start=20` - Return the next 20 users.

Considering your current implementation, it seems ideal to rename the `start` parameter to `filter`.